docs/reference/modules/discovery/discovery.asciidoc

```diff
@@ -0,0 +1,187 @@
+[[modules-discovery-hosts-providers]]
+=== Discovery
+
+The cluster formation module uses a list of _seed_ nodes in order to start off
+the discovery process. When you start an Elasticsearch node, or when a node
+believes the master node to have failed, that node tries to connect to each
+seed node in its list, and once connected the two nodes will repeatedly share
+information about the other known master-eligible nodes in the cluster in order
+to build a complete picture of the cluster.  By default the cluster formation
+module offers two hosts providers to configure the list of seed nodes: a
+_settings_-based and a _file_-based hosts provider. It can be extended to
+support cloud environments and other forms of hosts providers via
+{plugins}/discovery.html[discovery plugins]. Hosts providers are configured
+using the `discovery.zen.hosts_provider` setting, which defaults to the
+_settings_-based hosts provider. Multiple hosts providers can be specified as a
+list.
+
+[float]
+[[settings-based-hosts-provider]]
+===== Settings-based hosts provider
+
+The settings-based hosts provider use a node setting to configure a static list
+of hosts to use as seed nodes. These hosts can be specified as hostnames or IP
+addresses; hosts specified as hostnames are resolved to IP addresses during each
+round of discovery. Note that if you are in an environment where DNS resolutions
+vary with time, you might need to adjust your <<networkaddress-cache-ttl,JVM
+security settings>>.
+
+The list of hosts is set using the `discovery.zen.ping.unicast.hosts` static
```

Comment:
```suggestion
The list of hosts is set using the <<unicast.hosts,`discovery.zen.ping.unicast.hosts`>> static
```

Reply:
I think it's better to link to a reference page with all the lower-level details than cover it here.